File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Phasors/FNet/PhasorDefinition.vb
'*******************************************************************************************************
'  PhasorDefinition.vb - FNet Phasor definition
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/08/2007 - J. Ritchie Carroll & Jian (Ryan) Zuo
'       Initial version of source generated
'
'*******************************************************************************************************

Imports System.Runtime.Serialization

Namespace FNet

    <CLSCompliant(False), Serializable()> _
    Public Class PhasorDefinition

        Inherits PhasorDefinitionBase

        Protected Sub New()
        End Sub

        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)

            MyBase.New(info, context)

        End Sub

        Public Sub New(ByVal parent As ConfigurationCell)

            MyBase.New(parent)

        End Sub

        Public Sub New(ByVal parent As ConfigurationCell, ByVal index As Int32, ByVal label As String, ByVal scale As Int32, ByVal offset As Single, ByVal type As PhasorType, ByVal voltageReference As PhasorDefinition)

            MyBase.New(parent, index, label, scale, offset, type, voltageReference)

        End Sub

        Public Sub New(ByVal phasorDefinition As IPhasorDefinition)

            MyBase.New(phasorDefinition)

        End Sub

        ' FNet supports no configuration frame in the data stream - so there will be nothing to parse
        'Public Sub New(ByVal parent As ConfigurationCell, ByVal binaryImage As Byte(), ByVal startIndex As Int32)

        '    MyBase.New(parent, binaryImage, startIndex)

        'End Sub

        'Friend Shared Function CreateNewPhasorDefintion(ByVal parent As IConfigurationCell, ByVal binaryImage As Byte(), ByVal startIndex As Int32) As IPhasorDefinition

        '    Return New PhasorDefinition(parent, binaryImage, startIndex)

        'End Function

        Public Overrides ReadOnly Property DerivedType() As System.Type
            Get
                Return Me.GetType
            End Get
        End Property

        Public Shadows ReadOnly Property Parent() As ConfigurationCell
            Get
                Return MyBase.Parent
            End Get
        End Property

        ' All FNET values are floating point
        Public Overrides ReadOnly Property ScalePerBit() As Single
            Get
                Return 1.0F
            End Get
        End Property

    End Class

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/FNet/PhasorDefinition.cs
using System.Diagnostics;
using System;
//using TVA.Common;
using System.Collections;
using TVA.Interop;
using Microsoft.VisualBasic;
using TVA;
using System.Collections.Generic;
//using TVA.Interop.Bit;
using System.Linq;
using System.Runtime.Serialization;

//*******************************************************************************************************
//  PhasorDefinition.vb - FNet Phasor definition
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: VB.NET, Visual Studio 2008
//  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
//      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
//       Phone: 423/751-2827
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/08/2007 - J. Ritchie Carroll & Jian (Ryan) Zuo
//       Initial version of source generated
//
//*******************************************************************************************************


namespace PhasorProtocols
{
    namespace FNet
    {

        [CLSCompliant(false), Serializable()]
        public class PhasorDefinition : PhasorDefinitionBase
        {



            protected PhasorDefinition()
            {
            }

            protected PhasorDefinition(SerializationInfo info, StreamingContext context)
                : base(info, context)
            {


            }

            public PhasorDefinition(ConfigurationCell parent)
                : base(parent)
            {


            }

            public PhasorDefinition(ConfigurationCell parent, int index, string label, int scale, float offset, PhasorType type, PhasorDefinition voltageReference)
                : base(parent, index, label, scale, offset, type, voltageReference)
            {


            }

            public PhasorDefinition(ConfigurationCell parent, IPhasorDefinition phasorDefinition)
                : base(parent, phasorDefinition)
            {


            }

            // FNet supports no configuration frame in the data stream - so there will be nothing to parse
            //Public Sub New(ByVal parent As ConfigurationCell, ByVal binaryImage As Byte(), ByVal startIndex As int)

            //    MyBase.New(parent, binaryImage, startIndex)

            //End Sub

            //Friend Shared Function CreateNewPhasorDefinition(ByVal parent As IConfigurationCell, ByVal binaryImage As Byte(), ByVal startIndex As int) As IPhasorDefinition

            //    Return New PhasorDefinition(parent, binaryImage, startIndex)

            //End Function

            public override System.Type DerivedType
            {
                get
                {
                    return this.GetType();
                }
            }

            public new ConfigurationCell Parent
            {
                get
                {
                    return (ConfigurationCell)base.Parent;
                }
            }

        }

    }

}
